app/models/mediaflux/http/query_request.rb

```diff
@@ -29,28 +25,17 @@ def self.service
         "asset.query"
       end
 
-      # Parse the result of the query into a hash with a list of ids and a cursor to show
-      #  you where you are in the paginated results
+      # Returns the iterator that could be used to fetch the data
       def result
         xml = response_xml
-        ids = xml.xpath("/response/reply/result/id").children.map(&:text)
-
-        # total is only the actual total when the "complete" attribute is true,
-        # otherwise it reflects the total fetched so far
-        {
-          ids: ids,
-          size: xml.xpath("/response/reply/result/size").text.to_i,
-          cursor: parse_cursor(xml.xpath("/response/reply/result/cursor")),
-          files: parse_files(xml)
-        }
+        xml.xpath("/response/reply/result/iterator").text.to_i
```

Comment:
This is the "big" change. It returns an iterator (rather than a cursor). We then use the iterator to fetch the data via calls to `IteratorRequest` as shown in `app/models/project.rb` below.